server/src/main/java/org/elasticsearch/repositories/RepositoriesModule.java

```diff
@@ -102,7 +102,7 @@ public RepositoriesModule(
         }
         if (preRestoreChecks.isEmpty()) {
             preRestoreChecks.add((snapshot, version) -> {
-                if (version.isLegacyIndexVersion()) {
+                if (version.before(IndexVersions.MINIMUM_COMPATIBLE)) {
```

Comment:
Similar to the comment above, we will likely change this too to N - 2 later, but for now we should not allow indices created in N-2 to be restored.

Reply:
Done.